packages/core/core/src/BundleGraph.js

```diff
@@ -348,22 +348,40 @@ export default class BundleGraph {
     // Remove bundle node if it no longer has any entry assets
     let bundleNode = nullthrows(this._graph.getNode(bundle.id));
     if (this._graph.getNodesConnectedFrom(bundleNode).length === 0) {
-      let bundleGroupNodes = this._graph.getNodesConnectedTo(
-        bundleNode,
-        'bundle',
+      this.removeBundle(bundle);
+    }
+
+    this._bundleContentHashes.delete(bundle.id);
+  }
+
+  removeBundle(bundle: Bundle) {
+    // Remove bundle node if it no longer has any entry assets
+    let bundleNode = nullthrows(this._graph.getNode(bundle.id));
+
+    let bundleGroupNodes = this._graph.getNodesConnectedTo(
+      bundleNode,
+      'bundle',
+    );
+    this._graph.removeNode(bundleNode);
```

Comment:
Should we recursively remove child bundles that aren't reachable by any others? We currently only ever call this when deduplicating, in which case these bundles would be reachable, but there's nothing stopping this from being called in other cases. cc @padmaia @devongovett

Reply:
Not yet in the public API though right? So plugins wouldn't have access to this.